### PR TITLE
feat(VGMdb): add presence

### LINKED ISSUE #5435: VGMdb | vgmdb.net
### Discussed in https://github.com/PreMiD/Presences/discussions/4494

<div type='discussions-op-text'>

<sup>Originally posted by **Isa27-0** September 23, 2020</sup>
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 

**What's the name of the service?** VGMdb (I think it is an abbreviation for "Video Game Music Database",~~now it has more than game soundtracks~~): https://vgmdb.net

**What should the Presence display?** Service name, album you are viewing, publisher name and the time you're on the page 

**If possible, please provide a logo for the service (512x512 minimum)** 
I kind of did one of their Twitter profile picture, because it was a bit of a horrible quality
VGMdb

</div>

### PR DESCRIPTION
**Pull request by kyrie25**
#### Resolves #5435 

<details>
<summary> Proof of the creation working: </summary>
<br>

image
image
image
image
image

</details>

### CODE REVIEW

File: websites/V/VGMDB/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.6",
+  "author": {
+    "name": "Kyrie",
+    "id": "368399721494216706"
+  },
+  "service": "VGMDB",
+  "description": {
+    "en": "VGMdb is a community project dedicated to cataloging the music of visual arts and video games.",
+    "vi_VN": "VGMdb là dự án cộng đồng dành riêng cho việc lưu trữ âm nhạc và tranh ảnh của trò chơi điện tử."
```
Comment (Bas950) [inline websites/V/VGMDB/dist/metadata.json:10]: db in caps in service but not when mentioned in description... which is correct?

**Review by Bas950 — DISMISSED** [review]

File: websites/V/VGMDB/dist/metadata.json (hunk 1 above)
Comment (kyrie25) [reply websites/V/VGMDB/dist/metadata.json:10]: The one without the caps is correct I think, but I accidentally pushed with the folder name with all caps and have no idea how to change it.
I can make a new PR if you want

Comment (Bas950) [reply websites/V/VGMDB/dist/metadata.json:10]: just rename the folder in vscode and push it should just remove the old and add the new

Comment (kyrie25) [reply websites/V/VGMDB/dist/metadata.json:10]: I tried renaming but it didn't count as a change for some reason, will look into it ASAP

Comment (kyrie25) [reply websites/V/VGMDB/dist/metadata.json]: Done

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/V/VGMdb/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "id": "368399721494216706"
+  },
+  "service": "VGMdb",
+  "description": {
+    "en": "VGMdb is a community project dedicated to cataloging the music of visual arts and video games.",
+    "vi_VN": "VGMdb là dự án cộng đồng dành riêng cho việc lưu trữ âm nhạc và tranh ảnh của trò chơi điện tử."
+  },
+  "url": "vgmdb.net",
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/y0ZrNmX.png",
```
[the remark this answers is not in the document]
Comment (Isa27-0) [reply websites/V/VGMdb/dist/metadata.json:14]: > Very pixelated, could you find anything better?

In fact, this "logo" was made by me. At the time I hadn't found one in good condition for my Issue, so I improvised based on the [Twitter profile](https://mobile.twitter.com/vgmdb). There's only [this page](http://www.userlogos.org/logo/kibbas/05022009/vgmdbnet) from 2009 that contains the logo.

[the remark this answers is not in the document]
Comment (kyrie25) [reply websites/V/VGMdb/dist/metadata.json:14]: > Very pixelated, could you find anything better?

Guess I can try upscaling their Discord server icon, but it doesn't look any better itself tbh and is partially cropped, this is probably the best one I could find so far

[the remark this answers is not in the document]
Comment (kyrie25) [reply websites/V/VGMdb/dist/metadata.json:14]: This is the logo from their official GitHub also, it is not much better tbh
image

[the remark this answers is not in the document]
Comment (kyrie25) [reply websites/V/VGMdb/dist/metadata.json]: Nvm upscaled and changed, looks much better now

File: websites/V/VGMdb/presence.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    path = document.location.pathname;
+
+  if (path.startsWith("/album")) {
+    presenceData.details = document.querySelector<HTMLSpanElement>(
+      "#innermain > h1 > span:nth-child(1)"
+    ).textContent;
+    presenceData.state =
+      document.querySelector<HTMLDivElement>("#coverart").title;
+    if (cover) {
+      delete presenceData.largeImageKey;
```
Comment (Slowlife01) [inline websites/V/VGMdb/presence.ts:25]:
```suggestion
```

other lines too

File: websites/V/VGMdb/presence.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "logo",
+      startTimestamp: browsingTimestamp
+    },
+    cover = await presence.getSetting<boolean>("cover"),
+    shortTitle = document.title.substring(
+      0,
+      document.title.lastIndexOf("-") - 1
+    ),
```
Comment (Slowlife01) [inline websites/V/VGMdb/presence.ts:15]:
```suggestion
    shortTitle = document.title.match(/(.*) -/)[1],
```

File: websites/V/VGMdb/presence.ts (hunk 3 above)
Comment (kyrie25) [reply websites/V/VGMdb/presence.ts]: Done

**Review by Slowlife01 — APPROVED** [review]
lgtm

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**